modules/nf-plugin-template/src/main/nextflow/hello/HelloExtension.groovy

```diff
@@ -0,0 +1,155 @@
+package nextflow.hello
+
+import groovy.util.logging.Slf4j
+import groovyx.gpars.dataflow.DataflowReadChannel
+import groovyx.gpars.dataflow.DataflowWriteChannel
+import groovyx.gpars.dataflow.expression.DataflowExpression
+import nextflow.Channel
+import nextflow.Global
+import nextflow.Session
+import nextflow.extension.ChannelExtensionPoint
+import nextflow.extension.CH
+import nextflow.NF
+import nextflow.extension.DataflowHelper
+import nextflow.extension.Factory
+import nextflow.extension.Function
+import nextflow.extension.Operator
+import nextflow.hello.functions.HelloFunctions
+import nextflow.plugin.Scoped
+
+import java.util.concurrent.CompletableFuture
+
+/**
+ * @author : jorge <jorge.aguilera@seqera.io>
+ *
+ */
+@Slf4j
+@Scoped('hello')
+class HelloExtension extends ChannelExtensionPoint{
```

Comment:
Is this needed for testing purposes? would not be better to include into `test` source three of another submodule? or a [test fixture](https://docs.gradle.org/current/userguide/java_testing.html#declaring_dependencies_of_test_fixtures)?

Reply:
The idea of nf-plugin-template is to have a "full" plugin that can be tested as it.

In this way, we build the zip before running the test and the FakeIndex serves it as a typical plugin

Reply:
Understand that, but this will enter into the distribution I guess. Wondering if moving into test source tree of the nf-commons project would work the same?

Reply:
I think if we move to nf-commons we can run the typical tests calling functions, validating functions are detected, etc  but I think we'll not be able to test the plugin as integrated in a DSL 

If we decide is not required to test the DSL (at least via spock), probably we can remove the nf-plugin-template and move HelloExtension to nf-commons fixtures

Reply:
Not sure, I think it should work because when the plugin class in on the classpath it does not have to be loaded and started explicitly. This is indeed how it works during dev (see also [here](https://pf4j.org/doc/system-extension.html))

Reply:
Not sure how to do it.

I'm able in the test to do `def tmp = Plugins.getExtension(HelloExtension)` and it load the extension (once I've changed the build.gradle)

but the test always fails with `Unknown Nextflow plugin 'nf-plugin-template'` when `includeDef` try to load it so probably I'm missing something

Reply:
Umm, maybe it's needed to add the `test` path here. https://github.com/nextflow-io/nextflow/blob/2982-allow-the-import-for-custom-function-from-plugin-extensions/modules/nf-commons/src/main/nextflow/plugin/DevPluginClasspath.groovy#L38-L38

Reply:
Took me a while but I think I found the way to build and load the plugin

https://github.com/nextflow-io/nextflow/blob/c59ea8483b9d79305ff422926a6826b469ee5a2e/modules/nf-commons/src/test/nextflow/plugin/BuildPluginTrait.groovy

Reply:
cool!